pkg/apis/kops/kubeletconfig.go

```diff
@@ -56,5 +61,27 @@ func BuildKubeletConfigSpec(cluster *Cluster, instanceGroup *InstanceGroup) (*Ku
 		c.NodeLabels[k] = v
 	}
 
+	sv, err := util.ParseKubernetesVersion(cluster.Spec.KubernetesVersion)
+	if err != nil {
+		return c, fmt.Errorf("Failed to lookup kubernetes version: %v", err)
+	}
+
+	// --register-with-taints was available in the first 1.6.0 alpha, no need to rely on semver's pre/build ordering
+	sv.Pre = nil
+	sv.Build = nil
+	if sv.GTE(semver.Version{Major: 1, Minor: 6, Patch: 0, Pre: nil, Build: nil}) {
+		for i, t := range instanceGroup.Spec.Taints {
+			if c.Taints == nil {
+				c.Taints = make([]string, len(instanceGroup.Spec.Taints))
+			}
+			c.Taints[i] = t
+		}
+
+		// Enable scheduling since it can be controlled via taints.
+		// For pre-1.6.0 clusters, this is handled by taints.go
+		registerSchedulable := true
+		c.RegisterSchedulable = &registerSchedulable
+	}
```

Comment:
In that scenario, the taints would make it into the instance group spec, but would have no effect (ie wouldn't be passed on to kubelet). In other words, I've tried to leave pre-1.6 scenarios as they are (handled by tainter.go).

So I understand if we want to fail (or warn?) more loudly, rather than silently ignoring that field on pre-1.6. But I don't think we care about the actual contents in that case.